lib/log.c

```diff
@@ -55,7 +75,7 @@ const char *lookup_msg(const struct message *mz, int kz, const char *nf)
 	static struct message nt = {0};
 	const char *rz = nf ? nf : "(no message found)";
 	const struct message *pnt;
-	for (pnt = mz; memcmp(pnt, &nt, sizeof(struct message)); pnt++)
+	for (pnt = mz; message_cmp(pnt, &nt); pnt++)
```

Comment:
NAK, this code doesn't compare 2 messages to each other, it checks for the end of list terminator.  While your change doesn't break it, it is semantically wrong.

You can probably replace the `memcmp` with a `pnt->str != NULL` check instead.  (No point in having a message array entry without text… - but please double check that this is indeed the case.)